### PR TITLE
Fix third party auth

### PR DESCRIPTION
**Pull request by Fenntasy**
## Description

This PR updates the way the popup works to allow third party authentication.

Before, we opened an URL on Connect OAuth that opened an iframe with the authorization flow.
Now, we open directly the authorization flow by wrapping the state in a JSON object, converted to base64 to pass along the domain of the popup parent in order to let the last page of the authorization flow pass the authorization code to the parent.

## Motivation and Context

We want the popup to be seamless and to work as well as the normal flow.

## How Has This Been Tested?

Locally but intensely

## Types of changes

- [ ] Chore (non-breaking change which refactors / improves the existing code base)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My change requires a change to a package version.
- [x] I have updated the `package.json` version accordingly.
- [x] I have updated the `CHANGELOG.md` version accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by frieda-ewlines** [comment]
Task linked: [CU-1crxk8w Fix Connect Popup authentication with third party](https://app.clickup.com/t/1crxk8w)

File: popup/src/index.ts [hunk 1]
```diff
@@ -22,13 +22,18 @@ type Options = {
   onAuthorizationCodeReceived?: AuthorizationCodeCallback;
 };
 
+function utf8_to_b64(str: string): string {
+  return window.btoa(unescape(encodeURIComponent(str)));
+}
+
```
Comment (Fuzznimp) [inline popup/src/index.ts:28]: TIL `unescape`.

**Review by Fuzznimp — APPROVED** [review]

File: popup/src/index.ts (hunk 1 above)
Comment (Fenntasy) [reply popup/src/index.ts:28]: Yeah, it's from https://developer.mozilla.org/en-US/docs/Glossary/Base64#solution_1_–_escaping_the_string_before_encoding_it

Comment (Fuzznimp) [reply popup/src/index.ts:28]: Nice, I'll read it for sure!

**Pull request merged**